### ISSUE TITLE
Geometry representation of Features

### ISSUE THREAD

**Issue by remiColletGG** [Interoperable-data/ERA-Ontology-3.1.0#158]
#### TL;DR

GML/WKT are not able to represent curved lines. How should we report or represent the data, then ?
This issue pertains more to the micro (nano ?) level of detail.

#### Preamble

If we take an `era:Track` and go up its lineage, we discover it is a specialization of a `gsp:Feature`. As we can see, `gsp:Feature gsp:hasGeometry gsp:Geometry`. This means that a track has a representation in the physical world that can be pinpointed on a map, with the help of geo coordinates (a `gsp:Geometry`). Great !

#### WKT
Often, we represent such coordinates with the help of Well-Known Text (WKT). For example, a track could have this representation :
`
<http://www.opengis.net/def/crs/EPSG/0/31370> Line( {point}, {point}, {point}, {point} )^^geo:wktLiteral
`

Where :

* The first part tells us that the coordinates are in Lambert72 system (https://epsg.io/31370, that's the one used in Belgium, e.g.)
* The actual Line WKT with points coordinates that, _linked together by straight lines_, form a line
* The last part after ^^ informs us about the datatype (which is WKT, and not a xsd:double, for example)

#### GML
We can also use Geographic Markup Language (GML). It's more precise and structured in the xml fashion :
```
<Feature   fid="142" featureType="school"  Description="A middle school">
        <Polygon name="extent" srsName="epsg:27354">
            <LineString name="extent" srsName="epsg:27354">
                <CData>
                  491888.999999459,5458045.99963358 491904.999999458,5458044.99963358
                  491908.999999462,5458064.9996335 491924.999999461,5458064.99963358
                  491925.999999462,5458079.99963359 491977.999999466,5458120.9996336
                  491953.999999466,5458017.99963357 </CData>
            </LineString>
        </Polygon>
</Feature>
```
A bit more verbose than WKT, as you can see.

#### The data available 
In the micro (nano) level, tracks are certainly not straight lines. I would wager that the tracks plans are designed with the help of AutoCAD (or a similar software) that draws lines as functions. At Infrabel, our engineers extract LandXML files that the track builders can use.

Here is a sample of how a curved line section is described in the LandXML file : 
```
<Curve rot="cw" chord="131.468443065736" crvType="arc" delta="5.023333327932" dirEnd="234.543560222904" dirStart="239.566893550836" external="1.44240716763" length="131.510558994542" midOrd="1.441021474494" radius="1499.999999979075" tangent="65.797431874401">
	<Start>99399.241533075896 217417.628958541696</Start>
	<Center>100159.039614490655 216124.297262720531</Center>
	<End>99288.913853154008 217346.132423204777</End>
	<PI>99342.509596977179 217384.300450207753</PI>
</Curve>
```
As you can see, an arc-shaped line is characterized by quite a few attributes, to help get the shape right. Check the following image to have an idea of how these attributes help defining the arc : 
![image](https://github.com/user-attachments/assets/e19478a0-7bb4-415f-a037-c5aa0ef849a4)

 #### The issue
The three last sections serve to show that there is a clash between how the data can be represented (usually WKT or GML with linear interpolation between points), and how the data should be represented (few attributes needed to get the correct shape of an arc)

Taking a look at the standards : https://www.ogc.org/publications/standard/sfa/ and the discussions going on over at OGC https://github.com/opengeospatial/ogc-feat-geo-json/issues/35, it seems that lines with non-linear interpolation between points is not going to be with us anytime soon.

This begs the questions :
1. What is the level of fidelity that we want to represent the tracks with ? At the upper levels, it think it poses no problem, but on the micro (or nano) level, it might.
2. If we want maximum fidelity, then the usual forms of representation do no allow for an accurate one. Should we take the LandXML or any other plan files as input ?
3. Do we want to extend the OGC standard, and provide an extended form of GML or WKT that could represent the tracks with enhanced fidelity ?

For example : 
`
<http://www.opengis.net/def/crs/EPSG/0/31370> cadcurvestring ( 99718.905986842539 217628.309272655199 , 99310.948431326789 218015.112545670854 , 99687.404152321149 217597.587421578151 , pi ( 99703.756686205292 217612.331456315733 ) ; rot(ccw) ; chord ( 44.002246667793 ) ; crvType(arc) ; delta (4.485730322491 ) ; dirEnd (227.961038932858 ) ; dirStart ( 223.475308610368 ) ; external ( 0.431006369669 ) ; length ( 44.013486564313 ) ; midOrd ( 0.430676183179 ) )
`

**Comment by gatemezing** [comment]
DEar @remiColletGG , thank you for your question.
Before going deeper into answering your question, please have a look at this SoL in CHE: https://data-interop.era.europa.eu/describe#http%3A%2F%2Fdata.europa.eu%2F949%2FfunctionalInfrastructure%2FsectionsOfLine%2F2248581a3681260ad38323ec76e520999bfa5270.
You'll see that the `gsp:hasGeometry` points to a LINESTRING that can be rendered using a WKTMap (see image)
<img width="715" alt="track-Sol-CHE" src="https://github.com/user-attachments/assets/d10e03f7-dc06-4f91-a232-69cf25e608cc" />

**Comment by remiColletGG** [comment]
Thanks for pointing out the fact that, indeed, the current state of affairs is that IMs provide WKT linestrings that can be easily visualized on a map.

My question was more oriented toward the fact that these representations may be imprecise when zooming in. Perhaps we do not care for these small imprecisions ! But I think that it is healthy to ask ourselves the question.

**Comment by DavidFontaineTVS** [comment]
Example from Switzerland

<SectionOfLine ValidityDateStart="2023-12-09" Shape="LINESTRING (7.0641501 46.6085116,7.0650356 46.6078677,7.0683996 46.6053717,7.0686108 46.6052239,7.0688925 46.6050647,7.069123 46.6049575,7.0694633 46.6048246,7.0697668 46.6047257,7.0702204 46.6046265,7.0709496 46.6044912,7.0722682 46.6042571,7.0746452 46.6038338,7.0749693 46.6037791,7.0753898 46.603739,7.0759135 46.6037002,7.0762453 46.6036617,7.076559 46.6035826,7.0767579 46.603505,7.0769716 46.6033761,7.0795732 46.6019604,7.080144 46.6016968,7.0807067 46.6014647,7.0813072 46.6012606,7.0820394 46.6010605,7.0826497 46.600932,7.0832166 46.6008636,7.0848359 46.6007202,7.0854404 46.6006915,7.0859377 46.6006894,7.0866279 46.6007212,7.0873961 46.6007893,7.088129 46.600886,7.0884601 46.6009392,7.0887951 46.6010068,7.0891233 46.6011193,7.089386 46.6012542,7.0895913 46.6013924,7.0901221 46.6018591,7.090937 46.6025939,7.0913133 46.6029341,7.0914626 46.6030461,7.0916261 46.6031951,7.0918094 46.6033144,7.091998 46.6034121,7.0925733 46.603663,7.0938974 46.604169,7.0943116 46.6043043,7.0945632 46.6043644,7.0947731 46.6043983,7.0950314 46.6044099,7.0953291 46.6043964,7.097788 46.6042176,7.0986732 46.6041798)">
<SOLIMCode Value="5234"/>
<SOLLineIdentification Value="254"/>
<SOLOPStart Value="CH04067"/>
<SOLOPEnd Value="CH04088"/>
<SOLLength Value="3.071"/>
<SOLNature Value="10" OptionalValue="Regular SoL"/>

**Comment by gatemezing** [comment]
Yes, you are right. For now, if we could at least reach the point where all IMs can follow the same pattern as Switzerland, it could be a first nice step.
